### PR TITLE
OEP submodule refactoring

### PR DESCRIPTION
**Pull request by PiotrBanuba**
Tested on macOS

### CODE REVIEW

**Review by dmitrystoyanovsky — COMMENTED** [review]
in general comments are the same as for oep_desktop sample.

File: libraries/utils/glfw_utils/src/glfw_window.cpp [hunk 1]
```diff
@@ -129,12 +127,10 @@ void glfw_window::create_window(const std::string& title, GLFWwindow* share)
...
 {
-    bnb_error* error = nullptr;
-    bnb_effect_player_load_glad_functions((void*)glfwGetProcAddress, &error);
-    if (error) {
-        bnb_error_destroy(error);
-        throw std::runtime_error("gladLoadGLLoader error");
-    }
+#if BNB_OS_WINDOWS || BNB_OS_MACOS
+    // it's only need for use while working with dynamic libs
+    utility::load_glad_functions((GLADloadproc) glfwGetProcAddress);
```
Comment (nick-korotysh-bnb) [inline libraries/utils/glfw_utils/src/glfw_window.cpp:132]: is this function implemented somewhere in this example?
looks like this is C++ call from `<bnb/effect_player/utility.hpp>`, but this example about C API and it should not depend on Banuba SDK C++ API
there is corresponding C API function `bnb_effect_player_load_glad_functions()` in C API header `effect_player.h`

File: main.cpp [hunk 2]
```diff
@@ -50,31 +46,27 @@ void play_effect(const std::string& effect_name)
 
     // Create and run instance of camera, pass callback for frames
     // Callback for received frame from the camera
-    auto ef_cb = [&oep, render_t](std::shared_ptr<nv12_image> image) {
+    auto ef_cb = [&oep, render_t](pixel_buffer_sptr image) {
```
Comment (nick-korotysh-bnb) [inline main.cpp:49]: looks like you call something async, it is bad idea to pass local variable by reference into callback function

Comment (nick-korotysh-bnb) [inline main.cpp:49]: `render_t` is bad variable name (looks like type name due to `_t` suffix), but let's keep it as it was before

**Review by nick-korotysh-bnb — COMMENTED** [review]
looks fine, minor comments

File: main.cpp (hunk 2 above)
Comment (nick-korotysh-bnb) [reply main.cpp:49]: this is safe because it is in main.cpp, and this object will be alive, my bad
closing comments

File: cmake/utils.cmake [hunk 3]
```diff
@@ -4,6 +4,21 @@
 
 include(CMakeParseArguments)
 
+#
```
Comment (dmitrystoyanovsky) [inline cmake/utils.cmake:7]: Macro can be removed sine there are no points of usage.

**Review by dmitrystoyanovsky — APPROVED** [review]

**Pull request merged**